### DISCUSSION TITLE
"Image" data type in Airtable style? Connecting records and images

### DISCUSSION THREAD

**Discussion by artemdzyuba** [supabase/supabase#3121] [Questions] [answered]
Hi, I'm new to Supabase and PostgreSQL. It has very nice GUI to work with a mouse, much close to my Airtable experience.
How can I attach a photo to a record? In Airtable there's an **Attachment** type, where we can put any files just with a mouse or upload with API

In Supabase I haven't found any data type like that. If using Storages, how can I connect them to records in a correct way?

**Comment by dshukertjr** [comment, marked as answer]
Hi @artemdzyuba!

In order to attach an image to a record in your database, you can either store the URL or the path of your image on your database. 

You can upload an image with the [upload()](https://supabase.io/docs/reference/javascript/storage-from-upload) method. Once your upload it, you can either get the [getPublicUrl()](https://supabase.io/docs/reference/javascript/storage-from-getpublicurl) if your bucket is public and store the URL into the database, or for a privately shared image you can just store the path into the database and run [createSignedUrl()](https://supabase.io/docs/reference/javascript/storage-from-createsignedurl) to create a signed URL whenever your user needs to access the image.

**Reply by artemdzyuba** [reply]
Thank you!
